### PR TITLE
Fix #116, Fix #118

### LINKED ISSUE #116: Add properties/tags to runInfo objects?
Request from SA infrastructure partner. runInfo is a lightly specified object. Do we provide properties/tags here to put arbitrary content? Should we take a pass at trying to populate this element with useful information?

Two good examples: 
id: log files may need a key that associates the content with a larger automation run, to assist with coalescing results
platform: many automation jobs are repeated by hardware platform

### LINKED ISSUE #118: locations should be optional
Our spec requires a location despite noting that there could be exceptions. We should make this a SHOULD and emphasize that producers should provide this information in all cases where it is available and relevant. An example of where it may not be: a result that is a note with no rule id that simply provides logging style verbose msgs.

6.12.6. locations property

A result object shall contain a property named locations whose value is an array, each of whose elements is a location object (§6.13) which specifies a location where the result occurred. The locations array shall contain at least one element if the result can be associated with a location. In rare circumstances, it might not be possible to specify a location for a result.

### PR DESCRIPTION
**Pull request by ghost**
#116: Add properties/tags to runInfo objects?
#118: locations should be optional

Added properties and tags, as well as two first-class properties, correlationId and architecture.

Also:
- Specified a token to replace redacted information

@michaelcfanning

### CODE REVIEW

File: document/CoreFormat.mdk [hunk 1]
```diff
@@ -259,18 +259,6 @@ The information in the `runInfo` object makes it possible to precisely repeat a
...
-A `runInfo` object may contain a property named `runStartTime` whose value is a string
-specifying the date and time at which the run started.
-The string shall be in the format specified by (&sect;[#fileFormat-dateTime]).
-
-### `runEndTime` property { #runInfo-runEndTime }
-
-A `runInfo` object may contain a property named `runEndTime` whose value is a string
-specifying the date and time at which the run ended.
-The string shall be in the format specified by (&sect;[#fileFormat-dateTime]).
-
```
Comment (ghost) [inline document/CoreFormat.mdk:273]: Moved to the Tool Orchestration Profile.

File: document/Organization.mdk [hunk 2]
```diff
@@ -12,8 +12,11 @@ For example, a result management system might augment the information defined in
...
-along with three profiles:
-result management (&sect;[#result-management-profile]),
-policy compliance (&sect;[#policy-compliance-profile]),
-and rule metadata (&sect;[#rule-metadata-profile]).
+along with four profiles:
+
+* Result management (&sect;[#result-management-profile])
+* Policy compliance (&sect;[#policy-compliance-profile])
+* Rule metadata (&sect;[#rule-metadata-profile])
+* Tool orchestration (&sect;[#tool-orchestration-profile])
```
Comment (michaelcfanning) [inline document/Organization.mdk:20]: LGTM. We should ask ourselves whether we actually require both an orchestration and result mgmt. profile or if these things relate sufficiently for one. Can discuss later.

File: document/ToolOrchestrationProfile.mdk [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+The Tool Orchestration Profile defines four additional properties on the `runInfo` object (&sect;[#runInfo]).
+
+### `correlationId` property { #tool-orchestration-profile-runInfo-correlationId }
+
+A `runInfo` object may contain a property named `correlationId` whose value is a string
+that allows the run to be associated with other artifacts.
+
+~ IsoNote
+In an environment where a static analysis tool is executed as part of a larger
+automated build process, the "build id" might serve as the `correlationId`,
```
Comment (michaelcfanning) [inline document/ToolOrchestrationProfile.mdk:26]: Note that the correlationId can refer to a broader automation run, could be useful to cite this as an example. It seems odd to have the example focus on build.

File: document/ToolOrchestrationProfile.mdk [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+allowing the tool run to be associated with other artifacts produced by the build.
+~
+
+### `architecture` property { #tool-orchestration-profile-runInfo-architecture }
+
+A `runInfo` object may contain a property named `architecture` that specifies the hardware
```
Comment (michaelcfanning) [inline document/ToolOrchestrationProfile.mdk:32]: 'the hardware architecture at which the analysis targets are targeted'. a bit clumsy, use of target seems redundant.
how about
'the hardware architecture for which the analysis run was targeted' or 'against which'?

**Pull request merged**